cmd/bee/cmd/start.go

```diff
@@ -476,17 +476,29 @@ func (c *command) configureSigner(cmd *cobra.Command, logger log.Logger) (config
 
 	logger.Info("swarm public key", "public_key", hex.EncodeToString(crypto.EncodeSecp256k1PublicKey(publicKey)))
 
-	libp2pPrivateKey, created, err := keystore.Key("libp2p", password)
+	// use secp256r1 for libp2p as requires specific elliptic curve implementations only
+	libp2pPrivateKey, created, err := keystore.Key("libp2p", password, crypto.GenerateSecp256r1Key, crypto.EncodeSecp256r1PrivateKey, crypto.DecodeSecp256r1PrivateKey)
 	if err != nil {
-		return nil, fmt.Errorf("libp2p key: %w", err)
+		if !errors.Is(err, crypto.ErrX509Parse) {
+			return nil, fmt.Errorf("libp2p key: %w", err)
+		}
+
+		logger.Info("migrating the libp2p keys...")
```

Comment:
ideally any migration code should be encapsulated in individual function which would make it easier later on to identify unnecessary migration logic and also to remove it.